src/tools/rbd_mirror/ImageDeleter.cc

```diff
@@ -127,260 +124,186 @@ template <typename I>
 ImageDeleter<I>::ImageDeleter(ContextWQ *work_queue, SafeTimer *timer,
                               Mutex *timer_lock,
                               ServiceDaemon<librbd::ImageCtx>* service_daemon)
-  : m_work_queue(work_queue),
+  : m_work_queue(work_queue), m_timer(timer), m_timer_lock(timer_lock),
     m_service_daemon(service_daemon),
-    m_delete_lock("rbd::mirror::ImageDeleter::Delete"),
-    m_image_deleter_thread(this),
-    m_failed_timer(timer),
-    m_failed_timer_lock(timer_lock),
+    m_max_concurrent_deletions(g_ceph_context->_conf->get_val<uint64_t>(
+      "rbd_mirror_concurrent_image_deletions")),
+    m_failed_interval(g_ceph_context->_conf->get_val<double>(
+      "rbd_mirror_delete_retry_interval")),
+    m_lock("rbd::mirror::ImageDeleter::m_lock"),
```

Comment:
nit: If we did not cache these config settings in member variables we could change them in runtime.